Life.java

```diff
@@ -1,55 +1,168 @@
 public class Life implements ILife {
+
+  int [][] cellMatrix;
   
-  public static void main(String[] args) {
-    Life l = new Life(new String[] {  "     ",
+  public static void main(String[] args) 
+  {
+
+    // 5 Array Einträge á 5 Zeichen
+   /* Life l = new Life(new String[] {  "     ",
                                       "     ",
                                       " *** ",
                                       "     ",
                                       "     " });
     l = (Life) l.nextGeneration();
+    */
   }
 
-
   public Life() {
-    nukeAll();
-  }
-
-  public Life(String[] setup) {
-    this();
-    for (int y = 0; y < setup.length; y++)
-      for (int x = 0; x < setup[y].length(); x++)
+    this.cellMatrix = new int[5][5];
+    /*for (int y = 0; y < cellMatrix.length; y++){
+      for (int x = 0; x < cellMatrix.length; x++){
         if (setup[y].charAt(x) != ' ')
-          setAlive(x, y);
+          setAlive(x,y);
+      }
+    }
+    */
   }
 
 
   @Override
-  public void nukeAll() {
+  //setzt im gesamten Array alle Leerzeichen
+
     // TODO Auto-generated method stub
 
+    public void nukeAll() {
+    for(int y = 0; y < cellMatrix.length; y++){
+      for(int x = 0; x < cellMatrix.length; x++){
+        this.cellMatrix[x][y] = 0;
+      }
+    }
+
   }
 
+
   @Override
   public void setAlive(int x, int y) {
     // TODO Auto-generated method stub
-
+    if(!checkAlive(x,y)){
+      cellMatrix[x][y] = 1;
+    }
+    
   }
 
   @Override
   public void setDead(int x, int y) {
-    // TODO Auto-generated method stub
 
+    if(checkAlive(x,y)){
+      cellMatrix[x][y] = 0;
+    }
   }
 
-  @Override
-  public boolean isAlive(int x, int y) {
-    // TODO Auto-generated method stub
-    return false;
-  }
+
+  /*checkt ob an der Position y,x ein 
+  Leerzeichen ist wenn ja dann gibt die Methode
+  false zurück, sprich die Zelle ist nicht am leben
+  */
+  public boolean checkAlive(int x, int y) {
+    if(x < 0 || x >= 5){
+      return false;
+    }
+    if(y < 0 || y >= 5){
+      return false;
+    }
+    if (cellMatrix[x][y] != 0){
+      return true;
+    }
+		return false;
+	}
+
 
   @Override
   public ILife nextGeneration() {
-    // TODO Auto-generated method stub
+    // TODO Auto-generated method 
+    int[][] tempBoard = new int[5][5];
```

Comment:
Dadurch, dass sie die magischen Zahlen 5 überall in ihren Code vewenden, ist der Änderungsaufwand für eine andere Feldgröße relativ groß und sehr fehleranfällig, wenn Sie so etwas machen, dann bitte mit Konstanten arbeiten, oder wie hier, flexible Feldgrößen anbieten.